vertical-pod-autoscaler/pkg/recommender/logic/recommender_test.go

```diff
@@ -115,3 +117,53 @@ func TestControlledResourcesFilteredDefault(t *testing.T) {
 	assert.Contains(t, recommendedResources[containerName].LowerBound, model.ResourceCPU)
 	assert.Contains(t, recommendedResources[containerName].UpperBound, model.ResourceCPU)
 }
+
+func TestMapToListOfRecommendedContainerResources(t *testing.T) {
+	cases := []struct {
+		name         string
+		resources    RecommendedPodResources
+		expectedLast []string
+	}{
+		{
+			name: "All recommendations sorted",
+			resources: RecommendedPodResources{
+				"a-container": RecommendedContainerResources{Target: model.Resources{model.ResourceCPU: model.CPUAmountFromCores(1), model.ResourceMemory: model.MemoryAmountFromBytes(1000)}},
+				"b-container": RecommendedContainerResources{Target: model.Resources{model.ResourceCPU: model.CPUAmountFromCores(1), model.ResourceMemory: model.MemoryAmountFromBytes(1000)}},
+				"c-container": RecommendedContainerResources{Target: model.Resources{model.ResourceCPU: model.CPUAmountFromCores(1), model.ResourceMemory: model.MemoryAmountFromBytes(1000)}},
+				"d-container": RecommendedContainerResources{Target: model.Resources{model.ResourceCPU: model.CPUAmountFromCores(1), model.ResourceMemory: model.MemoryAmountFromBytes(1000)}},
+			},
+			expectedLast: []string{
+				"a-container",
+				"b-container",
+				"c-container",
+				"d-container",
+			},
+		},
+		{
+			name: "All recommendations unsorted",
+			resources: RecommendedPodResources{
+				"b-container": RecommendedContainerResources{Target: model.Resources{model.ResourceCPU: model.CPUAmountFromCores(1), model.ResourceMemory: model.MemoryAmountFromBytes(1000)}},
+				"a-container": RecommendedContainerResources{Target: model.Resources{model.ResourceCPU: model.CPUAmountFromCores(1), model.ResourceMemory: model.MemoryAmountFromBytes(1000)}},
+				"d-container": RecommendedContainerResources{Target: model.Resources{model.ResourceCPU: model.CPUAmountFromCores(1), model.ResourceMemory: model.MemoryAmountFromBytes(1000)}},
+				"c-container": RecommendedContainerResources{Target: model.Resources{model.ResourceCPU: model.CPUAmountFromCores(1), model.ResourceMemory: model.MemoryAmountFromBytes(1000)}},
+			},
+			expectedLast: []string{
+				"a-container",
+				"b-container",
+				"c-container",
+				"d-container",
+			},
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			namespace := "test-namespace"
+			vpa := model.NewVpa(model.VpaID{Namespace: namespace, VpaName: "my-vpa"}, labels.Nothing(), time.Unix(0, 0))
+			vpa.UpdateRecommendation(MapToListOfRecommendedContainerResources(tc.resources))
+			// Check that the slice is in the correct order.
+			for i := range vpa.Recommendation.ContainerRecommendations {
+				assert.Equal(t, tc.expectedLast[i], vpa.Recommendation.ContainerRecommendations[i].ContainerName)
```

Comment:
We could iterate over both index and item

```
for i, outRecommendation := range vpa.Recommendation.ContainerRecommendations {
```

and use the item in the assertion

```
assert.Equal(t, tc.expectedLast[i], outRecommendation)
```